src/Approximations/iterative_refinement.jl

```diff
@@ -1,82 +1,145 @@
 """
-    Approximation2D{N<:AbstractFloat}
+    LocalApproximation{N<:AbstractFloat}
 
 Type that represents a local approximation in 2D.
 
 ### Fields
 
-- `p1`        -- first inner point
-- `d1`        -- first direction
-- `p2`        -- second inner point
-- `d2`        -- second direction
-- `q`         -- intersection of the lines `x : d1.x = p1` and `x : d2.x = p2`
-- `err`       -- error made
-- `ndir`      -- normal direction of the inner approximation
-- `refinable` -- states if this approximation is refinable
+- `p1`         -- first inner point
+- `d1`         -- first direction
+- `p2`         -- second inner point
+- `d2`         -- second direction
+- `q`          -- intersection of the lines l1 ⟂ d1 at p1 and l2 ⟂ d2 at p2
+- `refinable`  -- states if this approximation is refinable
+- `err`        -- error upper bound
+
+### Notes
+
+The criteria for refinable is determined in the method `new_approx`.
 """
-struct Approximation2D{N<:AbstractFloat}
+struct LocalApproximation{N<:AbstractFloat}
```

Comment:
Why not ``N<:Real``? Because of the ``norm(..., 2)`` below?

Reply:
ok, discussed in gitter -- let's use `AbstractFloat` since the floating point tolerance `eps(N)` is needed for 0 crossings